### PR TITLE
Fixes fires [READY]

### LINKED ISSUE #42038: someone broke atmos again
https://i.gyazo.com/c3634a9afe6b6d864fb2fe1e50a2cc73.png

round id isnt relevant, screenshot is just to show the type ofthing that can happen easily. Engine wasnt sabotaged really, just turned on without being set up properly, it heated up insanely quickly though. Same thing happened in a round earlier today with the burn chamber for atmos, Get to extremely high temperatures (1e6k+) very quickly.

@4dplanner probably

### PR DESCRIPTION
**Pull request by 4dplanner**
:cl:
fix: fires will no longer burn quite so hot
fix: the first spark of a hotspot is no longer always the same temperature
/:cl:

I misunderstood the atmos code and fucked up. The part I used was for starting off small fires, and so was inappropriate for the big fires it was then applied to. This should fix it.
fixes #42038 

The effects of this pr **compared to just a revert** are:

A: a hotspot's initial burn will have its temperature set if created by hotspot_expose. Previously the first hit was always at the same temperature.
B: the initial burn is never bypassed. Bypassing meant that large volume hotspots like those created by igniters never actually increased the heat of the tile by more than the initial spark (which was weak due to A).

So if you used an igniter on some gas, it would
originally: weak burn, bypass, go out
broken pr: weak burn, strong burn, grow, grow....
This one: strong burn, bypass, spread fires to nearby tiles if necessary

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
i ded

**Comment by 4dplanner** [comment]
Note: currently improves the problem, but fires are still very (~30k) hot.

**Comment by kriskog** [comment]
If it actually works, please get a testmerge/speedmerge asap. SM is useless as is, always goes singulo and irradiates most of the station.

I walked into engineering on meta and died in less than 5 seconds just inside the door.

**Comment by Jordie0608** [comment]
This is testmerged, we'll see how it goes.

**Comment by davethwave** [comment]
Hope this doesn't destroy fusion

**Comment by 4dplanner** [comment]
> Hope this doesn't destroy fusion

Sorry for the slow response, but destroy fusion how? Compared to before I broke everything, there should only be some fires that burn a bit hotter due to fixing the bug that I didn't make. I'll clarify the OP.

File: code/modules/atmospherics/environmental/LINDA_fire.dm [hunk 1]
```diff
@@ -75,24 +77,19 @@
 
 	location.active_hotspot = src
 
-	if(volume > CELL_VOLUME*0.95)
-		bypassing = TRUE
-	else
-		bypassing = FALSE
+	bypassing = !just_spawned && (volume > CELL_VOLUME*0.95)
```
Comment (Cyberboss) [inline code/modules/atmospherics/environmental/LINDA_fire.dm]: This variable has [other possible effects](https://github.com/tgstation/tgstation/blob/master/code/modules/atmospherics/environmental/LINDA_fire.dm#L189). Is this still valid?

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/atmospherics/environmental/LINDA_fire.dm (hunk 1 above)
Comment (4dplanner) [reply code/modules/atmospherics/environmental/LINDA_fire.dm]: Note that the other effects are in process(), which returns early if just_spawned is true. As such, this is equivalent to the old method except that it won't just do nothing on the first bypass.

**Comment by YakumoChen** [comment]
SM almost killed me as a cyborg with the current state of the bug. I didn't even think cyborgs could take heat damage.

**Comment by 4dplanner** [comment]
> SM almost killed me as a cyborg with the current state of the bug. I didn't even think cyborgs could take heat damage.

Some fires do burn a bit hotter compared to before due to ignition being more effective, but I found them to be within reasonable ranges, but I'll look into it a bit more when I get back to my PC. How thoroughly was the SM sabotaged, and where were you standing when you got burned? Were you in a fire, next to the SM, or just elsewhere and burned by hot air?

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**